core/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java

```diff
@@ -219,12 +214,12 @@ protected IndexRoutingTable indexRoutingTable(ClusterState clusterState, String
 
     // either routing is set, or type/id are set
 
-    protected IndexShardRoutingTable shards(ClusterState clusterState, String index, String type, String id, String routing) {
-        int shardId = shardId(clusterState, index, type, id, routing);
+    protected IndexShardRoutingTable shards(ClusterState clusterState, String index, String id, String routing) {
+        int shardId = generateShardId(clusterState, index, id, routing);
         return shards(clusterState, index, shardId);
     }
 
-    protected IndexShardRoutingTable shards(ClusterState clusterState, String index, int shardId) {
+    public IndexShardRoutingTable shards(ClusterState clusterState, String index, int shardId) {
```

Comment:
can we call this shardRoutingTable?, if we expose it to the public, I wonder if we should just put `shardRoutingTable` method on the routing table , as a shortcut. Wouldn't it make more sense?